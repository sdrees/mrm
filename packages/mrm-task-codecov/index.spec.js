jest.mock('fs');
jest.mock('git-username');
jest.mock('mrm-core/src/util/log', () => ({
	added: jest.fn(),
}));

const { getConfigGetter } = require('mrm');
const vol = require('memfs').vol;
const task = require('./index');

const stringify = o => JSON.stringify(o, null, '  ');

const config = getConfigGetter({ github: 'gh' });

const travisYml = `language: node_js
node_js:
  - 8
`;
const packageJson = stringify({
	name: 'unicorn',
	scripts: {
		'test:coverage': 'jest --coverage',
	},
});
const readmeMd = '# Unicorn';

afterEach(() => vol.reset());

it('should add CodeCov', () => {
	vol.fromJSON({
		'/.travis.yml': travisYml,
		'/package.json': packageJson,
		'/Readme.md': readmeMd,
	});

	task(config);

	expect(vol.toJSON()).toMatchSnapshot();
});

it('should throw when .travis.yml not found', () => {
	vol.fromJSON({
		'/package.json': packageJson,
		'/Readme.md': readmeMd,
	});

	const fn = () => task(config);

	expect(fn).toThrowError('Run travis task');
});

it('should throw when coverage script not found', () => {
	vol.fromJSON({
		'/.travis.yml': travisYml,
		'/package.json': stringify({
			name: 'unicorn',
		}),
		'/Readme.md': readmeMd,
	});

	const fn = () => task(config);

	expect(fn).toThrowError('npm script not found');
});

it('should not throw when readme file not found', () => {
	vol.fromJSON({
		'/.travis.yml': travisYml,
		'/package.json': packageJson,
	});

	const fn = () => task(config);

	expect(fn).not.toThrowError();
});
