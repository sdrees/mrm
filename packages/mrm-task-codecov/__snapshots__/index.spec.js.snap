// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add CodeCov 1`] = `
Object {
  "/.travis.yml": "language: node_js
node_js:
  - 8
script:
  - npm run test -- --coverage
after_success:
  - 'bash <(curl -s https://codecov.io/bash)'
",
  "/Readme.md": "# Unicorn

[![Codecov](https://codecov.io/gh/gh/unicorn/branch/master/graph/badge.svg)](https://codecov.io/gh/gh/unicorn)",
  "/package.json": "{
  \\"name\\": \\"unicorn\\",
  \\"scripts\\": {
    \\"test:coverage\\": \\"jest --coverage\\"
  }
}",
}
`;
